src/lib/users/users_database.rb

```diff
@@ -53,33 +59,51 @@ def push(database)
     # @param root_dir [String] Path where the original "/" is mounted
     def self.import(root_dir)
       data = UsersDatabase.new
-      data.read_files(File.join(root_dir, "etc"))
-      return if data.passwd.nil? || data.passwd.empty?
+      data.read_files(root_dir)
+      if data.users.empty?
+        log.info "No users to import (#{root_dir})"
+        return
+      end
 
       push(data)
     end
 
+    def users
+      config&.users&.reject { |u| ignore?(u) } || []
+    end
+
+    def filtered_config(names)
+      filtered = config.clone
+
+      filtered.detach(*filtered.groups)
+      skipped = filtered.users.reject { |u| names.include?(u.name) }
+      filtered.detach(*skipped)
+
+      filtered
+    end
+
     # Populates the object with information read from a directory
     #
-    # @param dir [String] path to a directory containing passwd and shadow files
-    def read_files(dir)
+    # @param root_dir [String] Path where the original "/" is mounted
+    def read_files(root_dir)
+      dir = File.join(root_dir, "etc")
       passwd = File.join(dir, "passwd")
       shadow = File.join(dir, "shadow")
       return unless File.exist?(passwd) && File.exist?(shadow)
 
-      self.passwd = IO.read(passwd)
-      self.shadow = IO.read(shadow)
+      @config = Y2Users::Config.new
+      Y2Users::Linux::LocalReader.new(root_dir).read_to(config)
+      config.users.each { |u| u.source = :local }
```

Comment:
this in fact should be set in LocalReader. I forgot to have it there, but it makes sense and especially if we want that merge to Reader result.